File: Java/RecyclerView11BaseClassForActivityAdapter/app/src/main/java/dev/ogabek/recyclerview11_baseclassforactivityadapter/activity/MainActivity.java
package dev.ogabek.recyclerview11_baseclassforactivityadapter.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;

import java.util.List;

import dev.ogabek.recyclerview11_baseclassforactivityadapter.R;
import dev.ogabek.recyclerview11_baseclassforactivityadapter.adapter.MainRecyclerViewAdapter;
import dev.ogabek.recyclerview11_baseclassforactivityadapter.model.Member;

public class MainActivity extends BaseActivity {

    Context context;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        refreshAdapter(prepareMemberList());

    }

    private void initViews() {
        context = this;
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context, 1));
    }

    private void refreshAdapter(List<Member> members) {
        MainRecyclerViewAdapter adapter = new MainRecyclerViewAdapter(context, members);
        recyclerView.setAdapter(adapter);
    }

}

File: Kotlin/RecyclerView11BaseClassForActivityAndAdapter/app/src/main/java/dev/ogabek/recyclerview_11_baseclassforactivityandadapter/activity/MainActivity.kt
package dev.ogabek.recyclerview_11_baseclassforactivityandadapter.activity

import android.content.Context
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview_11_baseclassforactivityandadapter.R
import dev.ogabek.recyclerview_11_baseclassforactivityandadapter.adapter.MainAdapter
import dev.ogabek.recyclerview_11_baseclassforactivityandadapter.model.Member

class MainActivity : BaseActivity() {

    private lateinit var context: Context
    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        refreshAdapter(prepareMemberList())

    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
    }

    private fun refreshAdapter(members: List<Member>) {
        val adapter = MainAdapter(this, members)
        recyclerView.adapter = adapter
    }

}